run-ws.py

```diff
@@ -0,0 +1,121 @@
+import os
+import json
+import socket
+import asyncio
+import logging
+import argparse
+import websockets
+
+from jsonschema import validate
+from chainlink import Chainlink
+
+import grader.api_keys as api_keys
+
+from config import *
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger()
+
+grading_job_def = {
+    "type": "object",
+    "properties": {
+        api_keys.GRADING_JOB_ID: {"type": "string"},
+        api_keys.STAGES: {"type": "array"},
+    },
+    "required": [api_keys.GRADING_JOB_ID, api_keys.STAGES],
+    "additionalProperties": False,
+}
+
+
+async def run(token, worker_id):
+    url = "{}://{}:{}{}{}/{}".format(
+        "wss" if USE_SSL else "ws",
+        API_HOSTNAME, API_PORT,
+        API_PROXY, WORKER_WS_ENDPOINT,
+        worker_id,
+    )
+
+    headers = {api_keys.AUTH: "Bearer {}".format(token)}
+    hostname = socket.gethostname()
+
+    async with websockets.connect(url, extra_headers=headers) as ws:
+        # poll job
+        try:
+            await ws.send(json.dumps({
+                "type": "register",
+                "args": {"hostname": hostname},
+            }))
+
+            ack = json.loads(await ws.recv())
+            assert ack["success"]
+
+            while True:
+                job = json.loads(await ws.recv())
+
+                validate(instance=job, schema=grading_job_def)
```

Comment:
`on_close` is the default event handler for connection loss in tornado. But as you said, I should explicitly send ping messages instead of relying on this event handler.